packages/steps/src/scripts/runCustomFunction.ts

```diff
@@ -0,0 +1,116 @@
+import assert from 'assert';
+
+import { createLogger } from '@expo/logger';
+import { SpawnPromise, SpawnResult } from '@expo/spawn-async';
+import cloneDeep from 'lodash.clonedeep';
+
+import { BuildStepOutput } from '../BuildStepOutput.js';
+import { BuildStepInput } from '../BuildStepInput.js';
+import { SerializedCustomBuildFunctionArguments } from '../utils/customFunction.js';
+import { BuildStepContext } from '../BuildStepContext.js';
+import { BuildStepFunction } from '../BuildStep.js';
+import { spawnAsync } from '../utils/shell/spawn.js';
+
+async function runCustomJsFunctionAsync(): Promise<void> {
+  const customJavascriptFunctionModulePath = process.argv[2];
+  const functionArgs = process.argv[3];
+
+  assert(customJavascriptFunctionModulePath, 'customJavascriptFunctionModulePath is required');
+  assert(functionArgs, 'serializedFunctionParams is required');
+
+  let serializedFunctionArguments: SerializedCustomBuildFunctionArguments;
+  try {
+    serializedFunctionArguments = JSON.parse(functionArgs);
+  } catch (e) {
+    console.error('Failed to parse serializedFunctionParams');
+    throw e;
+  }
+
+  const logger = createLogger({
+    name: 'customJavscriptFunctionLogger',
+    streams: [
+      {
+        type: 'raw',
+        stream: {
+          write: (rec: any) => {
+            if (rec) {
+              switch (rec.level) {
+                case 30: // Info level
+                  if (rec.msg) {
+                    console.log(rec.msg);
+                  }
+                  break;
+                case 40: // Warn level
+                  if (rec.msg) {
+                    console.warn(rec.msg);
+                  }
+                  break;
+                case 50: // Error level
+                case 60: // Fatal level
+                  if (rec.msg) {
+                    console.error(rec.msg);
+                  }
+                  break;
+                default:
+                  break;
```

Comment:
Is it possible to log on a different level? Should we handle that by falling back to `console.debug/info` or something?